src/common/transformations/src/transformations/common_optimizations/fuse_rotary_positional_embeddings.cpp

```diff
@@ -463,10 +495,24 @@ ov::pass::RoPEFusionChatGLM::RoPEFusionChatGLM(int split_output_id) {
     auto var_split_2 = makePattern<opset1::VariadicSplit>({cos_sin_cache, 0, {0, ov::gen_pattern::Symbol("end")}});
     var_split_2->set_output_size(2);
 
-    auto view_Reshape_460 =
-        makePattern<opset1::Reshape>({slice_StridedSlice_449 | slice_Slice_449 | var_split_2->output(0),
-                                      ListConstruct_379_Concat | const_target_shape_2},
-                                     {{"special_zero", false}});
+    std::shared_ptr<ov::Node> view_Reshape_460 = nullptr;
+    if (support_2d_rope) {
+        // Slice cos_sin_cache to support 2-dimentional RoPE
+        auto ScatterUpdate = makePattern<opset3::ScatterUpdate>({{0, 0}, {1}, seq_length, {0}}, {});
+        auto slice_Slice_449 = makePattern<ov::opset8::Slice>({cos_sin_cache, {0,0}, ScatterUpdate, {1,1}, {0}});
```

Comment:
`slice_Slice_449` could also be replaced with 1-D parameter with axis 1
```
auto slice_Slice_449 = makePattern<ov::opset8::Slice>({cos_sin_cache, {0}, seq_length, {1}, {1}}, {}, nullptr, "cos_sin_slice");
```

Reply:
In chatglm4 graph, slice_449's inputs have 2 dimension like this graph, how could it match pattern with 1d parameter?

![slice_449](https://github.com/user-attachments/assets/b611919c-1d57-4905-90e7-9431ae03cf40)

Reply:
Fixed it. thanks.